mindsdb_native/libs/controllers/predictor.py

```diff
@@ -87,22 +82,23 @@ def prepare_sample_settings(self,
         sample_settings['sample_function'] = sample_settings['sample_function'].__name__
         return sample_settings, sample_function
 
+    @mdb_lock(flags=portalocker.LOCK_EX+portalocker.LOCK_NB, lock_name='learn', argname='self.name')
     def learn(self,
               to_predict,
               from_data,
               test_from_data=None,
               group_by=None,
               window_size=None,
               order_by=None,
+              sample_margin_of_error=0.005,
               ignore_columns=None,
               stop_training_in_x_seconds=None,
               backend='lightwood',
               rebuild_model=True,
               use_gpu=None,
               equal_accuracy_for_all_output_categories=True,
               output_categories_importance_dictionary=None,
-              unstable_parameters_dict=None,
-              sample_settings=None):
+              unstable_parameters_dict=None):
```

Comment:
Careful, something is wrong. You are removing `sample_settings` and bringing back `sample_margin_of_error`